components/list/list-item-drag-mixin.js

```diff
@@ -1,20 +1,77 @@
+import { css, html } from 'lit-element/lit-element.js';
 import { announce } from '../../helpers/announce.js';
+import { dragActions } from './list-item-drag-handle.js';
 import { getUniqueId } from '../../helpers/uniqueId.js';
-import { html } from 'lit-element/lit-element.js';
 import { nothing } from 'lit-html';
 
+const move = Object.freeze({
+	above: true,
+	below: false
+});
+
+const timeDelayForLeavingTheDropArea = 1000; //ms
+
 export const ListItemDragMixin = superclass => class extends superclass {
 
 	static get properties() {
 		return {
 			draggable: { type: Boolean, reflect: true },
-			key: { type: String, reflect: true }
+			dragText: { type: String, attribute: 'drag-text' },
+			key: { type: String, reflect: true },
+			beingDragged: { type: Boolean, reflect: true, attribute: 'being-dragged' },
+			_bottomPlacementMarker: { type: Boolean },
+			_dropTarget: { type: Boolean },
```

Comment:
This triggers whether or not to show the special drop area that gives better usability.

Reply:
It would be nice to not need to keep track of drop targets in the state map (`_visitedDestination`).  Some other options to consider:

* only conditionally rendering based on `draggable`, or
* reset `_dropTarget` in `dragleave` or `dragexit` events, thereby avoiding holding references in state object

If we really need to conditionally render, perhaps consider renaming this `_dropTarget` property since it is currently named as an object/element rather than a boolean.  Perhaps `_draggingOver` would be better.

Reply:
I really tired to make dragexit or dragleave work. Sadly, I ran into problems with the mouse being moved to quickly. This is why I decided for a central holder of the state. It also allows us to extend the 1 second border. Your other comments below will make it better. 

I did keep the commit around for when I did use drag leave. It worked, but there were edge cases that were wonky.

This comment has similar information: https://github.com/BrightspaceUI/core/pull/665#discussion_r449159726

Reply:
Yeah maybe complicated by the setTimeout...?  I'd expect the event handlers to be triggered in a consistent order, and synchronously.